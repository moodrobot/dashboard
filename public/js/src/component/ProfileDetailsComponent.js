import React, { Component } from "react";
import TextComponent from "../component/TextComponent";
import DropDownComponent from "../component/DropDownComponent";
import SubHead from "../component/SubHead";
import PropTypes from "prop-types";

class ProfileDetailsComponent extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    //console.log("RENDERING PROFILE DETAILS");

    const TT = this.props.translateTitle;
    const TE = this.props.translateError;
    const CE = this.props.componentHasError;

    return (
      <div>
        <SubHead title={TT["title"]} />
        <TextComponent
          title={TT["first-name"]}
          id="first_name"
          hasError={CE.first_name.flag}
          required="Required"
          error_message={TE[CE.first_name.message]}
          size="small"
          value={this.props.data.first_name}
          handler={this.props.handleProfileChange}
          disabled={this.props.freezeAll}
        />
        <TextComponent
          title={TT["last-name"]}
          id="last_name"
          hasError={CE.last_name.flag}
          required="Required"
          size="small"
          error_message={TE[CE.last_name.message]}
          value={this.props.data.last_name}
          handler={this.props.handleProfileChange}
          disabled={this.props.freezeAll}
        />
        <TextComponent
          title={TT["display-name"]}
          id="display_name"
          hasError={CE.display_name.flag}
          required="Required"
          size="small"
          error_message={TE[CE.display_name.message]}
          value={this.props.data.display_name}
          handler={this.props.handleProfileChange}
          disabled={this.props.freezeAll}
        />
        <TextComponent
          title={TT["email"]}
          id="email"
          size="small"
          hasError={CE.email.flag}
          required="Required"
          error_message={TE[CE.email.message]}
          value={this.props.data.email}
          handler={this.props.handleProfileChange}
          disabled={this.props.freezeAll}
        />
        <TextComponent
          title={TT["company"]}
          id="company"
          size="small"
          hasError={CE.company.flag}
          error_message={TE[CE.company.message]}
          value={this.props.data.company}
          handler={this.props.handleProfileChange}
          disabled={this.props.freezeAll}
        />
        <TextComponent
          title={TT["phone-number"]}
          id="phone"
          size="small"
          hasError={CE.phone.flag}
          error_message={TE[CE.phone.message]}
          value={this.props.data.phone}
          handler={this.props.handleProfileChange}
          disabled={this.props.freezeAll}
        />
        <DropDownComponent
          title={TT["country"]}
          data={this.props.countries}
          id="country"
          defaultValue={this.props.data.country}
          onChangeHandler={this.props.handleProfileChange}
          value={this.props.data.country}
          disabled={this.props.freezeAll}
        />
        <DropDownComponent
          title={TT["language-preference"]}
          //data={t("profile-details-language-preference-options", {
          //  returnObjects: true
          //})}
          data={TT["language-preference-options"]}
          id="language"
          value={this.props.currentLanguage}
          defaultValue={this.props.data.language}
          onChangeHandler={this.props.handleLanguageChange}
          disabled={this.props.freezeAll}
        />
        <TextComponent
          title={TT["about"]}
          id="bio"
          hasError={CE.bio.flag}
          error_message={TE[CE.bio.message]}
          type="area"
          row={6}
          align_top="true"
          value={this.props.data.bio}
          handler={this.props.handleProfileChange}
          disabled={this.props.freezeAll}
        />
        <TextComponent
          title={TT["signature"]}
          id="signature"
          hasError={CE.signature.flag}
          error_message={TE[CE.signature.message]}
          value={this.props.data.signature}
          handler={this.props.handleProfileChange}
          disabled={this.props.freezeAll}
        />
        <SubHead title={TT["connect-title"]} />
        <TextComponent
          title={TT["connect-twitter"]}
          id="twitter"
          hasError={CE.twitter.flag}
          error_message={TE[CE.twitter.message]}
          nest="other_services"
          value={(this.props.data.other_services["twitter"]) ? (this.props.data.other_services["twitter"]) : ""}
          handler={this.props.handleProfileChange}
          disabled={this.props.freezeAll}
        />
        <TextComponent
          title={TT["connect-github"]}
          id="github"
          hasError={CE.github.flag}
          error_message={TE[CE.github.message]}
          nest="other_services"
          value={(this.props.data.other_services.github) ? (this.props.data.other_services.github) : ""}
          handler={this.props.handleProfileChange}
          disabled={this.props.freezeAll}
        />
        <TextComponent
          title={TT["connect-developer-mesh"]}
          id="developer-mesh"
          hasError={CE["developer-mesh"].flag}
          error_message={TE[CE["developer-mesh"].message]}
          nest="other_services"
          value={(this.props.data.other_services["developer-mesh"]) ? (this.props.data.other_services["developer-mesh"]) : ""}
          handler={this.props.handleProfileChange}
          disabled={this.props.freezeAll}
        />
        <TextComponent
          title={TT["connect-website"]}
          id="website"
          hasError={CE.website.flag}
          error_message={TE[CE.website.message]}
          nest="other_services"
          value={(this.props.data.other_services.website) ? (this.props.data.other_services.website) : ""}
          handler={this.props.handleProfileChange}
          disabled={this.props.freezeAll}
        />
      </div>
    );
  }
}

ProfileDetailsComponent.propTypes = {
  data: PropTypes.object,
  freezeAll: PropTypes.bool,
  translateTitle: PropTypes.object.isRequired,
  translateError: PropTypes.object.isRequired,
  countries: PropTypes.array,
  componentHasError: PropTypes.object,
  handleProfileChange: PropTypes.func,
  handleLanguageChange: PropTypes.func,
  currentLanguage: PropTypes.string,
  init: PropTypes.bool
};

export default ProfileDetailsComponent;
