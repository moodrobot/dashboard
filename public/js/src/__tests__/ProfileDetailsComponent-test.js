import React, { Component } from "react";
import { shallow } from "enzyme";
import sinon from "sinon";
import ProfileDetailsComponent from "../component/ProfileDetailsComponent.js";
import profileData from "./__mockData__/profileData.json";
import countries from "./__mockData__/countries.json";
import translateData from "../../../../locales/en-US.json";
import i18n from "../i18n-for-testing"; // initialized i18next instance for testing

let componentHasError = {
  first_name: { flag: false, message: "" },
  last_name: { flag: false, message: "" },
  display_name: { flag: false, message: "" },
  email: { flag: false, message: "" },
  phone: { flag: false, message: "" },
  company: { flag: false, message: "" },
  bio: { flag: false, message: "" },
  signature: { flag: false, message: "" },
  "developer-mesh": { flag: false, message: "" },
  twitter: { flag: false, message: "" },
  github: { flag: false, message: "" },
  website: { flag: false, message: "" }
};

let allERROR = {
  first_name: { flag: true, message: "noHTML" },
  last_name: { flag: true, message: "noHTML" },
  display_name: { flag: true, message: "noHTML" },
  email: { flag: true, message: "noHTML" },
  phone: { flag: true, message: "noHTML" },
  company: { flag: true, message: "noHTML" },
  bio: { flag: true, message: "noHTML" },
  signature: { flag: true, message: "noHTML" },
  "developer-mesh": { flag: true, message: "noHTML" },
  twitter: { flag: true, message: "noHTML" },
  github: { flag: true, message: "noHTML" },
  website: { flag: true, message: "noHTML" }
};

describe("<ProfileDetailsComponent />", () => {
  const wrapper = shallow(
    <ProfileDetailsComponent
      data={profileData[0]}
      translateTitle={i18n.t("profile-details")}
      translateError={i18n.t("error")}
      componentHasError={componentHasError}
      countries={countries}
      handleProfileChange={function() {}}
      handleLanguageChange={function() {}}
    />
  );
  it("expect to match snapshot", () => {
    /**EXPECT NO CHANGE IN THE SNAPSHOT */
    expect(wrapper).toMatchSnapshot();

    /**CHECK PROPS TO MAKE SURE ARE MAPPING CORRECTLY - (INSTEAD OF SNAPSHOT?) */
    expect(
      wrapper
        .find("input")
        .first()
        .props().value === profileData[0].first_name
    );
  });
  it("expect setProps to update value with new data", () => {
    /**ADD DIFFERENT PROPS TO SIMULATE CHANGE FROM ABOVE */
    wrapper.setProps({ data: profileData[1] });

    wrapper.update();

    /** EXPECT PROPS TO UPDATE WITH NEW NAME */
    expect(
      wrapper
        .find("input")
        .first()
        .props().value === profileData[1].first_name
    );
  });
});
