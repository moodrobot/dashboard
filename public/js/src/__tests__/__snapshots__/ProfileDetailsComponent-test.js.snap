// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfileDetailsComponent /> expect to match snapshot 1`] = `
<div>
  <SubHead
    title="Profile Details"
  />
  <TextComponent
    handler={[Function]}
    hasError={false}
    id="first_name"
    required="Required"
    size="small"
    title="First Name"
    value="Brent"
  />
  <TextComponent
    handler={[Function]}
    hasError={false}
    id="last_name"
    required="Required"
    size="small"
    title="Last Name"
    value="Rieck"
  />
  <TextComponent
    handler={[Function]}
    hasError={false}
    id="display_name"
    required="Required"
    size="small"
    title="Display Name (Public)"
    value="Brent Rieck (Intel)"
  />
  <TextComponent
    handler={[Function]}
    hasError={false}
    id="email"
    required="Required"
    size="small"
    title="Email Address"
    value="brent.rieck@intel.com"
  />
  <TextComponent
    handler={[Function]}
    hasError={false}
    id="company"
    size="small"
    title="Company"
    value="Intel"
  />
  <TextComponent
    handler={[Function]}
    hasError={false}
    id="phone"
    size="small"
    title="Phone Number"
    value="503.503.5400"
  />
  <DropDownComponent
    data={
      Array [
        Object {
          "name": "Afghanistan",
          "value": "AF",
        },
        Object {
          "name": "Åland Islands",
          "value": "AX",
        },
        Object {
          "name": "Albania",
          "value": "AL",
        },
        Object {
          "name": "Algeria",
          "value": "DZ",
        },
        Object {
          "name": "American Samoa",
          "value": "AS",
        },
        Object {
          "name": "Andorra",
          "value": "AD",
        },
        Object {
          "name": "Angola",
          "value": "AO",
        },
        Object {
          "name": "Anguilla",
          "value": "AI",
        },
        Object {
          "name": "Antarctica",
          "value": "AQ",
        },
        Object {
          "name": "Antigua and Barbuda",
          "value": "AG",
        },
        Object {
          "name": "Argentina",
          "value": "AR",
        },
        Object {
          "name": "Armenia",
          "value": "AM",
        },
        Object {
          "name": "Aruba",
          "value": "AW",
        },
        Object {
          "name": "Australia",
          "value": "AU",
        },
        Object {
          "name": "Austria",
          "value": "AT",
        },
        Object {
          "name": "Azerbaijan",
          "value": "AZ",
        },
        Object {
          "name": "Bahamas",
          "value": "BS",
        },
        Object {
          "name": "Bahrain",
          "value": "BH",
        },
        Object {
          "name": "Bangladesh",
          "value": "BD",
        },
        Object {
          "name": "Barbados",
          "value": "BB",
        },
        Object {
          "name": "Belarus",
          "value": "BY",
        },
        Object {
          "name": "Belgium",
          "value": "BE",
        },
        Object {
          "name": "Belize",
          "value": "BZ",
        },
        Object {
          "name": "Benin",
          "value": "BJ",
        },
        Object {
          "name": "Bermuda",
          "value": "BM",
        },
        Object {
          "name": "Bhutan",
          "value": "BT",
        },
        Object {
          "name": "Bolivia",
          "value": "BO",
        },
        Object {
          "name": "Bosnia and Herzegovina",
          "value": "BA",
        },
        Object {
          "name": "Botswana",
          "value": "BW",
        },
        Object {
          "name": "Bouvet Island",
          "value": "BV",
        },
        Object {
          "name": "Brazil",
          "value": "BR",
        },
        Object {
          "name": "British Indian Ocean Territory",
          "value": "IO",
        },
        Object {
          "name": "Brunei Darussalam",
          "value": "BN",
        },
        Object {
          "name": "Bulgaria",
          "value": "BG",
        },
        Object {
          "name": "Burkina Faso",
          "value": "BF",
        },
        Object {
          "name": "Burundi",
          "value": "BI",
        },
        Object {
          "name": "Cambodia",
          "value": "KH",
        },
        Object {
          "name": "Cameroon",
          "value": "CM",
        },
        Object {
          "name": "Canada",
          "value": "CA",
        },
        Object {
          "name": "Cape Verde",
          "value": "CV",
        },
        Object {
          "name": "Cayman Islands",
          "value": "KY",
        },
        Object {
          "name": "Central African Republic",
          "value": "CF",
        },
        Object {
          "name": "Chad",
          "value": "TD",
        },
        Object {
          "name": "Chile",
          "value": "CL",
        },
        Object {
          "name": "China",
          "value": "CN",
        },
        Object {
          "name": "Christmas Island",
          "value": "CX",
        },
        Object {
          "name": "Cocos (Keeling) Islands",
          "value": "CC",
        },
        Object {
          "name": "Colombia",
          "value": "CO",
        },
        Object {
          "name": "Comoros",
          "value": "KM",
        },
        Object {
          "name": "Congo",
          "value": "CG",
        },
        Object {
          "name": "Congo, The Democratic Republic of the",
          "value": "CD",
        },
        Object {
          "name": "Cook Islands",
          "value": "CK",
        },
        Object {
          "name": "Costa Rica",
          "value": "CR",
        },
        Object {
          "name": "Cote D'ivoire",
          "value": "CI",
        },
        Object {
          "name": "Croatia",
          "value": "HR",
        },
        Object {
          "name": "Cuba",
          "value": "CU",
        },
        Object {
          "name": "Cyprus",
          "value": "CY",
        },
        Object {
          "name": "Czech Republic",
          "value": "CZ",
        },
        Object {
          "name": "Denmark",
          "value": "DK",
        },
        Object {
          "name": "Djibouti",
          "value": "DJ",
        },
        Object {
          "name": "Dominica",
          "value": "DM",
        },
        Object {
          "name": "Dominican Republic",
          "value": "DO",
        },
        Object {
          "name": "East Timor",
          "value": "TL",
        },
        Object {
          "name": "Ecuador",
          "value": "EC",
        },
        Object {
          "name": "Egypt",
          "value": "EG",
        },
        Object {
          "name": "El Salvador",
          "value": "SV",
        },
        Object {
          "name": "Equatorial Guinea",
          "value": "GQ",
        },
        Object {
          "name": "Eritrea",
          "value": "ER",
        },
        Object {
          "name": "Estonia",
          "value": "EE",
        },
        Object {
          "name": "Ethiopia",
          "value": "ET",
        },
        Object {
          "name": "Falkland Islands (Malvinas)",
          "value": "FK",
        },
        Object {
          "name": "Faroe Islands",
          "value": "FO",
        },
        Object {
          "name": "Fiji",
          "value": "FJ",
        },
        Object {
          "name": "Finland",
          "value": "FI",
        },
        Object {
          "name": "France",
          "value": "FR",
        },
        Object {
          "name": "French Guiana",
          "value": "GF",
        },
        Object {
          "name": "French Polynesia",
          "value": "PF",
        },
        Object {
          "name": "French Southern Territories",
          "value": "TF",
        },
        Object {
          "name": "Gabon",
          "value": "GA",
        },
        Object {
          "name": "Gambia",
          "value": "GM",
        },
        Object {
          "name": "Georgia",
          "value": "GE",
        },
        Object {
          "name": "Germany",
          "value": "DE",
        },
        Object {
          "name": "Ghana",
          "value": "GH",
        },
        Object {
          "name": "Gibraltar",
          "value": "GI",
        },
        Object {
          "name": "Greece",
          "value": "GR",
        },
        Object {
          "name": "Greenland",
          "value": "GL",
        },
        Object {
          "name": "Grenada",
          "value": "GD",
        },
        Object {
          "name": "Guadeloupe",
          "value": "GP",
        },
        Object {
          "name": "Guam",
          "value": "GU",
        },
        Object {
          "name": "Guatemala",
          "value": "GT",
        },
        Object {
          "name": "Guernsey",
          "value": "GG",
        },
        Object {
          "name": "Guinea",
          "value": "GN",
        },
        Object {
          "name": "Guinea-Bissau",
          "value": "GW",
        },
        Object {
          "name": "Guyana",
          "value": "GY",
        },
        Object {
          "name": "Haiti",
          "value": "HT",
        },
        Object {
          "name": "Heard Island and Mcdonald Islands",
          "value": "HM",
        },
        Object {
          "name": "Holy See (Vatican City State)",
          "value": "VA",
        },
        Object {
          "name": "Honduras",
          "value": "HN",
        },
        Object {
          "name": "Hong Kong",
          "value": "HK",
        },
        Object {
          "name": "Hungary",
          "value": "HU",
        },
        Object {
          "name": "Iceland",
          "value": "IS",
        },
        Object {
          "name": "India",
          "value": "IN",
        },
        Object {
          "name": "Indonesia",
          "value": "ID",
        },
        Object {
          "name": "Iran, Islamic Republic of",
          "value": "IR",
        },
        Object {
          "name": "Iraq",
          "value": "IQ",
        },
        Object {
          "name": "Ireland",
          "value": "IE",
        },
        Object {
          "name": "Isle of Man",
          "value": "IM",
        },
        Object {
          "name": "Israel",
          "value": "IL",
        },
        Object {
          "name": "Italy",
          "value": "IT",
        },
        Object {
          "name": "Jamaica",
          "value": "JM",
        },
        Object {
          "name": "Japan",
          "value": "JP",
        },
        Object {
          "name": "Jersey",
          "value": "JE",
        },
        Object {
          "name": "Jordan",
          "value": "JO",
        },
        Object {
          "name": "Kazakhstan",
          "value": "KZ",
        },
        Object {
          "name": "Kenya",
          "value": "KE",
        },
        Object {
          "name": "Kiribati",
          "value": "KI",
        },
        Object {
          "name": "Korea, Democratic People's Republic of",
          "value": "KP",
        },
        Object {
          "name": "Korea, Republic of",
          "value": "KR",
        },
        Object {
          "name": "Kuwait",
          "value": "KW",
        },
        Object {
          "name": "Kyrgyzstan",
          "value": "KG",
        },
        Object {
          "name": "Lao People's Democratic Republic",
          "value": "LA",
        },
        Object {
          "name": "Latvia",
          "value": "LV",
        },
        Object {
          "name": "Lebanon",
          "value": "LB",
        },
        Object {
          "name": "Lesotho",
          "value": "LS",
        },
        Object {
          "name": "Liberia",
          "value": "LR",
        },
        Object {
          "name": "Libya",
          "value": "LY",
        },
        Object {
          "name": "Liechtenstein",
          "value": "LI",
        },
        Object {
          "name": "Lithuania",
          "value": "LT",
        },
        Object {
          "name": "Luxembourg",
          "value": "LU",
        },
        Object {
          "name": "Macau",
          "value": "MO",
        },
        Object {
          "name": "Macedonia",
          "value": "MK",
        },
        Object {
          "name": "Madagascar",
          "value": "MG",
        },
        Object {
          "name": "Malawi",
          "value": "MW",
        },
        Object {
          "name": "Malaysia",
          "value": "MY",
        },
        Object {
          "name": "Maldives",
          "value": "MV",
        },
        Object {
          "name": "Mali",
          "value": "ML",
        },
        Object {
          "name": "Malta",
          "value": "MT",
        },
        Object {
          "name": "Marshall Islands",
          "value": "MH",
        },
        Object {
          "name": "Martinique",
          "value": "MQ",
        },
        Object {
          "name": "Mauritania",
          "value": "MR",
        },
        Object {
          "name": "Mauritius",
          "value": "MU",
        },
        Object {
          "name": "Mayotte",
          "value": "YT",
        },
        Object {
          "name": "Mexico",
          "value": "MX",
        },
        Object {
          "name": "Micronesia,  Federated States of",
          "value": "FM",
        },
        Object {
          "name": "Moldova, Republic of",
          "value": "MD",
        },
        Object {
          "name": "Monaco",
          "value": "MC",
        },
        Object {
          "name": "Mongolia",
          "value": "MN",
        },
        Object {
          "name": "Montenegro",
          "value": "ME",
        },
        Object {
          "name": "Montserrat",
          "value": "MS",
        },
        Object {
          "name": "Morocco",
          "value": "MA",
        },
        Object {
          "name": "Mozambique",
          "value": "MZ",
        },
        Object {
          "name": "Myanmar",
          "value": "MM",
        },
        Object {
          "name": "Namibia",
          "value": "NA",
        },
        Object {
          "name": "Nauru",
          "value": "NR",
        },
        Object {
          "name": "Nepal",
          "value": "NP",
        },
        Object {
          "name": "Netherlands",
          "value": "NL",
        },
        Object {
          "name": "Netherlands Antilles",
          "value": "AN",
        },
        Object {
          "name": "New Caledonia",
          "value": "NC",
        },
        Object {
          "name": "New Zealand",
          "value": "NZ",
        },
        Object {
          "name": "Nicaragua",
          "value": "NI",
        },
        Object {
          "name": "Niger",
          "value": "NE",
        },
        Object {
          "name": "Nigeria",
          "value": "NG",
        },
        Object {
          "name": "Niue",
          "value": "NU",
        },
        Object {
          "name": "Norfolk Island",
          "value": "NF",
        },
        Object {
          "name": "Northern Mariana Islands",
          "value": "MP",
        },
        Object {
          "name": "Norway",
          "value": "NO",
        },
        Object {
          "name": "Oman",
          "value": "OM",
        },
        Object {
          "name": "Pakistan",
          "value": "PK",
        },
        Object {
          "name": "Palau",
          "value": "PW",
        },
        Object {
          "name": "Palestinian Territory, Occupied",
          "value": "PS",
        },
        Object {
          "name": "Panama",
          "value": "PA",
        },
        Object {
          "name": "Papua New Guinea",
          "value": "PG",
        },
        Object {
          "name": "Paraguay",
          "value": "PY",
        },
        Object {
          "name": "Peru",
          "value": "PE",
        },
        Object {
          "name": "Philippines",
          "value": "PH",
        },
        Object {
          "name": "Pitcairn",
          "value": "PN",
        },
        Object {
          "name": "Poland",
          "value": "PL",
        },
        Object {
          "name": "Portugal",
          "value": "PT",
        },
        Object {
          "name": "Puerto Rico",
          "value": "PR",
        },
        Object {
          "name": "Qatar",
          "value": "QA",
        },
        Object {
          "name": "Reunion",
          "value": "RE",
        },
        Object {
          "name": "Romania",
          "value": "RO",
        },
        Object {
          "name": "Russian Federation",
          "value": "RU",
        },
        Object {
          "name": "Rwanda",
          "value": "RW",
        },
        Object {
          "name": "Saint Barthélemy",
          "value": "BL",
        },
        Object {
          "name": "Saint Helena",
          "value": "SH",
        },
        Object {
          "name": "Saint Kitts and Nevis",
          "value": "KN",
        },
        Object {
          "name": "Saint Lucia",
          "value": "LC",
        },
        Object {
          "name": "Saint Martin (French part)",
          "value": "MF",
        },
        Object {
          "name": "Saint Pierre and Miquelon",
          "value": "PM",
        },
        Object {
          "name": "Saint Vincent and The Grenadines",
          "value": "VC",
        },
        Object {
          "name": "Samoa",
          "value": "WS",
        },
        Object {
          "name": "San Marino",
          "value": "SM",
        },
        Object {
          "name": "Sao Tome and Principe",
          "value": "ST",
        },
        Object {
          "name": "Saudi Arabia",
          "value": "SA",
        },
        Object {
          "name": "Senegal",
          "value": "SN",
        },
        Object {
          "name": "Serbia",
          "value": "RS",
        },
        Object {
          "name": "Seychelles",
          "value": "SC",
        },
        Object {
          "name": "Sierra Leone",
          "value": "SL",
        },
        Object {
          "name": "Singapore",
          "value": "SG",
        },
        Object {
          "name": "Slovakia",
          "value": "SK",
        },
        Object {
          "name": "Slovenia",
          "value": "SI",
        },
        Object {
          "name": "Solomon Islands",
          "value": "SB",
        },
        Object {
          "name": "Somalia",
          "value": "SO",
        },
        Object {
          "name": "South Africa",
          "value": "ZA",
        },
        Object {
          "name": "South Georgia and The South Sandwich Islands",
          "value": "GS",
        },
        Object {
          "name": "South Sudan",
          "value": "SS",
        },
        Object {
          "name": "Spain",
          "value": "ES",
        },
        Object {
          "name": "Sri Lanka",
          "value": "LK",
        },
        Object {
          "name": "Sudan",
          "value": "SD",
        },
        Object {
          "name": "Suriname",
          "value": "SR",
        },
        Object {
          "name": "Svalbard and Jan Mayen",
          "value": "SJ",
        },
        Object {
          "name": "Swaziland",
          "value": "SZ",
        },
        Object {
          "name": "Sweden",
          "value": "SE",
        },
        Object {
          "name": "Switzerland",
          "value": "CH",
        },
        Object {
          "name": "Syrian Arab Republic",
          "value": "SY",
        },
        Object {
          "name": "Taiwan",
          "value": "TW",
        },
        Object {
          "name": "Tajikistan",
          "value": "TJ",
        },
        Object {
          "name": "Tanzania, United Republic of",
          "value": "TZ",
        },
        Object {
          "name": "Thailand",
          "value": "TH",
        },
        Object {
          "name": "Togo",
          "value": "TG",
        },
        Object {
          "name": "Tokelau",
          "value": "TK",
        },
        Object {
          "name": "Tonga",
          "value": "TO",
        },
        Object {
          "name": "Trinidad and Tobago",
          "value": "TT",
        },
        Object {
          "name": "Tunisia",
          "value": "TN",
        },
        Object {
          "name": "Turkey",
          "value": "TR",
        },
        Object {
          "name": "Turkmenistan",
          "value": "TM",
        },
        Object {
          "name": "Turks and Caicos Islands",
          "value": "TC",
        },
        Object {
          "name": "Tuvalu",
          "value": "TV",
        },
        Object {
          "name": "Uganda",
          "value": "UG",
        },
        Object {
          "name": "Ukraine",
          "value": "UA",
        },
        Object {
          "name": "United Arab Emirates",
          "value": "AE",
        },
        Object {
          "name": "United Kingdom",
          "value": "GB",
        },
        Object {
          "name": "United States",
          "value": "US",
        },
        Object {
          "name": "United States Minor Outlying Islands",
          "value": "UM",
        },
        Object {
          "name": "Uruguay",
          "value": "UY",
        },
        Object {
          "name": "Uzbekistan",
          "value": "UZ",
        },
        Object {
          "name": "Vanuatu",
          "value": "VU",
        },
        Object {
          "name": "Venezuela",
          "value": "VE",
        },
        Object {
          "name": "Vietnam",
          "value": "VN",
        },
        Object {
          "name": "Virgin Islands, British",
          "value": "VG",
        },
        Object {
          "name": "Virgin Islands, U.S.",
          "value": "VI",
        },
        Object {
          "name": "Wallis and Futuna",
          "value": "WF",
        },
        Object {
          "name": "Western Sahara",
          "value": "EH",
        },
        Object {
          "name": "Yemen",
          "value": "YE",
        },
        Object {
          "name": "Zambia",
          "value": "ZM",
        },
        Object {
          "name": "Zimbabwe",
          "value": "ZW",
        },
      ]
    }
    defaultValue="US"
    id="country"
    onChangeHandler={[Function]}
    title="Country/Region"
    value="US"
  />
  <DropDownComponent
    data={
      Array [
        Object {
          "name": "English",
          "value": "en-US",
        },
        Object {
          "name": "简体中文",
          "value": "zh-CN",
        },
        Object {
          "name": "Español",
          "value": "es-ES",
        },
        Object {
          "name": "Português",
          "value": "pt-BR",
        },
        Object {
          "name": "Русский",
          "value": "ru-RU",
        },
      ]
    }
    defaultValue="en-US"
    id="language"
    onChangeHandler={[Function]}
    title="Language Preference"
  />
  <TextComponent
    align_top="true"
    handler={[Function]}
    hasError={false}
    id="bio"
    row={6}
    title="About (Public)"
    type="area"
    value="All about Brent - he likes Astoria.

<b>yay?</b>"
  />
  <TextComponent
    handler={[Function]}
    hasError={false}
    id="signature"
    title="Signature (Public)"
    value="Signature"
  />
  <SubHead
    title="Connect (Public)"
  />
  <TextComponent
    handler={[Function]}
    hasError={false}
    id="twitter"
    nest="other_services"
    title="Twitter"
    value="bsr"
  />
  <TextComponent
    handler={[Function]}
    hasError={false}
    id="github"
    nest="other_services"
    title="GitHub"
    value="bsr"
  />
  <TextComponent
    handler={[Function]}
    hasError={false}
    id="developer-mesh"
    nest="other_services"
    title="Developer Mesh"
    value="bsr"
  />
  <TextComponent
    handler={[Function]}
    hasError={false}
    id="website"
    nest="other_services"
    title="Website"
    value="http://bsr.fun"
  />
</div>
`;
